### PR TITLE
EZP-32256: GraphQL field validation throws exception without 'eng-GB' language

### PR DESCRIPTION
**Pull request by ITernovtsiy**
Jira ticket: https://issues.ibexa.co/browse/EZP-32256

Shortly, this PR address 3 issues with validation
- `$fieldErrors['eng-GB']` is not an array, but single error
- use translated message text instead of `Value for required field definition '%identifier%' with language '%code%' is empty"`
- undefined index 'eng-GB' if no such language in the system

*STR:*
1. On clean install, replace `eng-GB` language to `eng-US`. You can use SQL to speed it up (instead of doing it manually in admin):
```
UPDATE `ezcontentclass`
SET `serialized_name_list` = REPLACE(`serialized_name_list`, "eng-GB", "eng-US")
WHERE `serialized_name_list` LIKE "%eng-GB%";

UPDATE `ezcontentclass`
SET `serialized_description_list` = REPLACE(`serialized_description_list`, "eng-GB", "eng-US")
WHERE `serialized_description_list` LIKE "%eng-GB%";

UPDATE `ezcontentclass_attribute`
SET `serialized_name_list` = REPLACE(`serialized_name_list`, "eng-GB", "eng-US")
WHERE `serialized_name_list` LIKE "%eng-GB%";

UPDATE `ezcontentclass_attribute`
SET `serialized_description_list` = REPLACE(`serialized_description_list`, "eng-GB", "eng-US")
WHERE `serialized_description_list` LIKE "%eng-GB%";

UPDATE `ezcontentclass_name`
SET `language_locale` = "eng-US"
WHERE `language_locale`="eng-GB";

UPDATE ezcontentobject_attribute
SET `language_code` = "eng-US"
WHERE `language_code`="eng-GB";

UPDATE ezcontentobject_attribute
SET `data_text` = REPLACE(`data_text`, "eng-GB", "eng-US")
WHERE `data_text` LIKE "%eng-GB%";

UPDATE ezcontentobject_name
SET `content_translation`="eng-US", `real_translation`="eng-US"
WHERE `content_translation`="eng-GB";

UPDATE `ezcontent_language`
SET `locale` ="eng-US", `name`="English (American)"
WHERE `locale`="eng-GB";
```

2. Regenerate graphql schema
3. Send graphql request with invalid field, for example:
```
mutation createUser {
  createUser(parentLocationId: 2, language: eng_US, input: {
    firstName: ""
    lastName: "test l"
    userAccount: "test@test.test"
  }) {
    id
  }
}
```

*AR:*
PHP Notice message returned:
```
"errors": [
    {
      "debugMessage": "Notice: Undefined index: eng-GB",
      "message": "Internal server Error",
```

*ER:*
Message about field validation fails returned:
```
"errors": [
    {
      "message": "Field 'first_name' failed validation: Value for required field definition 'first_name' with language 'eng-US' is empty",
```

### CODE REVIEW

File: src/GraphQL/Resolver/DomainContentMutationResolver.php [hunk 1]
```diff
@@ -277,20 +277,13 @@ static function (FieldDefinition $fieldDefinition) use ($fieldDefId) {
...
-                    if (count($fieldError['values']) === 1) {
-                        $message = $translatableMessage->singular;
-                    } else {
-                        $message = $translatableMessage->plural;
-                    }
-                } else {
-                    $message = $translatableMessage->message;
-                }
+            $fieldErrorsList = $fieldErrors['eng-GB'] ?? reset($fieldErrors);
+            $fieldErrorsList = is_array($fieldErrorsList) ? $fieldErrorsList : [$fieldErrorsList];
```
Comment (adamwojs) [inline src/GraphQL/Resolver/DomainContentMutationResolver.php:281]: Your patch is correct, however the root cause of the issue is hard-coded `eng-GB` language code, which shouldn't happen here :disappointed: 

Can we change `renderFieldValidationErrors`  implementation to:

```php
    private function renderFieldValidationErrors(RepositoryExceptions\ContentFieldValidationException $e, API\Values\ContentType\ContentType $contentType)
    {
        $errors = [];
        foreach ($e->getFieldErrors() as $fieldDefId => $fieldErrorsByLanguage) {
            $fieldDefinition = $contentType->getFieldDefinitions()->filter(
                static function (FieldDefinition $fieldDefinition) use ($fieldDefId) {
                    return $fieldDefinition->id === $fieldDefId;
                }
            )->first();

            foreach ($fieldErrorsByLanguage as $fieldErrors) {
                foreach ($fieldErrors as $fieldError) {
                    // depending on error instance, values injected in Plural::__toString or Message::__toString
                    $errors[] = sprintf("Field '%s' failed validation: %s",
                        $fieldDefinition->identifier,
                        (string)$fieldError->getTranslatableMessage()
                    );
                }
            }
        }

        return $errors;
    } 
```

// cc @ezsystems/php-dev-team @bdunogier

Comment (ITernovtsiy) [reply src/GraphQL/Resolver/DomainContentMutationResolver.php]: hm, I rechecked the flow again and I think we don't need to support array in `$fieldErrors` at all.
`$e->getFieldErrors()` should return this structure:
https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/Base/Exceptions/ContentFieldValidationException.php#L26

What do you think about variant I just pushed?

**Review by bdunogier — APPROVED** [review]
Hi, and thank you very much for your PR Ivan.

While we must still get rid of the hardcoded eng-GB, your patch is a good improvement for the time being.

**Review by lserwatka — APPROVED** [review]

**Review by micszo — APPROVED** [review]
Reproduced and retested on Ibexa Experience 3.2.3 with diff.

**Comment by lserwatka** [comment]
@bdunogier could you merge it up?

**Comment by bdunogier** [comment]
@lserwatka done for `master` and `3.0`.

**Pull request merged**